File: msg-sentiment/src/test/java/dev/number6/sentiment/dagger/TestChannelMessageSentimentComprehensionComponent.java
package dev.number6.sentiment.dagger;


import dev.number6.comprehend.results.PresentableSentimentResults;
import dev.number6.message.ChannelMessagesToComprehensionResultsFunction;
import dev.number6.message.ComprehensionResultsConsumer;
import dagger.Component;

import javax.inject.Singleton;

@Component(modules = {
        FakeCommonModule.class,
        ChannelMessagesHandlerModule.class,
        FakeComprehensionResultsModule.class
})
@Singleton
public interface TestChannelMessageSentimentComprehensionComponent extends ChannelMessagesSentimentComprehensionComponent {

    @Singleton
    ChannelMessagesToComprehensionResultsFunction<PresentableSentimentResults> getResultsFunction();

    @Singleton
    ComprehensionResultsConsumer<PresentableSentimentResults> getConsumer();

}

File: msg-sentiment/src/test/kotlin/dev/number6/sentiment/dagger/TestChannelMessageSentimentComprehensionComponent.kt
package dev.number6.sentiment.dagger

import dagger.Component
import dev.number6.comprehend.results.PresentableSentimentResults
import dev.number6.message.ChannelMessagesToComprehensionResultsFunction
import dev.number6.message.ComprehensionResultsConsumer
import javax.inject.Singleton

@Component(modules = [FakeComprehendModule::class, ChannelMessagesHandlerModule::class, FakeComprehensionResultsModule::class])
@Singleton
interface TestChannelMessageSentimentComprehensionComponent : ChannelMessagesSentimentComprehensionComponent {
    @Singleton
    fun getResultsFunction(): ChannelMessagesToComprehensionResultsFunction<PresentableSentimentResults>

    @Singleton
    fun getConsumer(): ComprehensionResultsConsumer<PresentableSentimentResults>
}